org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/jdt/JdtSourceLookUpProvider.java

```diff
@@ -98,6 +99,22 @@ public String getSourceFileURI(String fullyQualifiedName, String sourcePath) {
         if (fullyQualifiedName == null) {
             throw new IllegalArgumentException("fullyQualifiedName is null");
         }
+        // Jdt Search Engine doesn't support searching anonymous class or local type directly.
+        // For inner type, such as HelloWorld$InnerType, replace the separator "$" with "." and then search it.
+        // For anonymous type, such as HelloWorld$1, search its enclosing type instead.
+        if (fullyQualifiedName.indexOf("$") >= 0) {
+            String result = searchDeclarationFileByFqn(fullyQualifiedName.replace('$', '.'));
+            if (result != null) {
+                return result;
+            } else {
+                return searchDeclarationFileByFqn(AdapterUtils.parseEnclosingType(fullyQualifiedName));
+            }
+        } else {
+            return searchDeclarationFileByFqn(fullyQualifiedName);
+        }
+    }
+
+    private String searchDeclarationFileByFqn(String fullyQualifiedName) {
```

Comment:
>private String searchDeclarationFileByFqn(String fullyQualifiedName) { [](start = 4, length = 70)

move  private methods under public methods section?

Reply:
not required, I will add checkstyle and modify then